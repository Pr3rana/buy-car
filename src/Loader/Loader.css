.loader-list{
    display: flex;
    max-height: 100px;
    border: 1px solid #ededed;
    margin: 12px;
    align-items: center;
}

.loader-list-container p{
    margin-left: 24px;
    width: 150px;
    padding: 8px;
    /* margin-bottom: 0; */
    /* margin-top: 0; */
    background-color: #ededed;
    animation-name: fadeInFadeOut;
    animation-duration: 4s;
    animation-delay: 1s;
    animation-iteration-count: 3;
    animation-direction: alternate;
}
.loader-list-container .product-icon{
    margin: 12px;
    height: 80px;
    width: 80px;
    background-color: #ededed;
    animation-name: fadeInFadeOut;
    animation-duration: 4s;
    animation-delay: 1s;
    animation-iteration-count: 3;
    animation-direction: alternate;
}

.list-header h3{
    margin: 0 12px 24px;
}
.list-header p{
    margin: 0 12px 24px;
}

.list-content{
    display: inline-block;
}

.loader-pagination-container{
    text-align: center;
    margin-bottom: 24px;
}
.loader-pagination-container span{
    display: inline-block;
    width: 20%;
    padding: 12px;
    background-color: #ededed;
    animation-name: fadeInFadeOut;
    animation-duration: 4s;
    animation-delay: 1s;
    animation-iteration-count: 3;
    animation-direction: alternate;
}
@keyframes fadeInFadeOut {
    from {background-color: #ededed;}
    to {background-color: #474747;}
  }
/* 
.product-icon{
    padding: 18px;
    height: 100px;
    width: 100px;
} */