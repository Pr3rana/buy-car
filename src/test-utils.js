import React from 'react';
import { render } from '@testing-library/react'
import { BrowserRouter as Router, Route, Switch, Link } from 'react-router-dom'; 
import { ListPageContext } from "./helpers/appContext";

export const carData = {
    cars: [{"stockNumber":10000,"manufacturerName":"Tesla","modelName":"Model S","color":"silver","mileage":{"number":151884,"unit":"km"},"fuelType":"Diesel","pictureUrl":"https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"}],
    totalPageCount:100,
    totalCarsCount:1000
}
export const isFetched = false;
const setTotalCarsCount = jest.fn();
const setTotalPageCount = jest.fn();
const setPagenumber = jest.fn();
const setSelectedFilter = jest.fn();
const setSavedCars = jest.fn();
const setAvailableFilters = jest.fn();
const setFilterParams = jest.fn();

const pageContextProvider = {totalCarsCount:1,
    setTotalCarsCount,
    totalPageCount:2,
    setTotalPageCount,
    pagenumber:1,
    setPagenumber,
    selectedFilter:[{}],
    setSelectedFilter,
    savedCars:[],
    setSavedCars,
    availableFilters:{test:["test value"]},
    setAvailableFilters,
    filterParams: [{test:"test filter value"}],
    setFilterParams
}

const AllTheProviders = ({ children }) => {
    return (
     <Router>
          <ListPageContext.Provider value={pageContextProvider}>{children}</ListPageContext.Provider>
     </Router>
    )
}

const customRender = (ui, options) => {
    render(ui, { wrapper: AllTheProviders, ...options });
};

export * from "@testing-library/react";
export { customRender as render };