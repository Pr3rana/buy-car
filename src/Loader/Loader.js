import { useContext } from 'react';
// import {Link} from 'react-router-dom';
import Pagination from '../Pagination/Pageination';
import { ListPageContext } from "../helpers/storeContext";
import './Loader.css';

const Loader = ({ cars }) => {
    const { pagenumber } = useContext(ListPageContext);
    return ( 
        <div>
            <div className="listContainer list-wrapper">
                <div>
                    <h3>Available Cars</h3>
                    <p>Showing {pagenumber*10} out of 100 results</p>
                </div>
                {cars.map((car,index)=>(
                    <div className="list" key={index}>
                        <img alt="alt-text" className="product-icon" />
                        <div className="listContent">
                            <h3></h3>
                            <p className="product-key-details">
                                <span></span>
                                <span></span>
                                <span></span>
                                <span></span>
                            </p>
                            <p className="deatils-link">View details</p>
                        </div>
                    </div>
                ))}
                <Pagination />
            </div>
        </div>
    );
}
 
export default Loader;
