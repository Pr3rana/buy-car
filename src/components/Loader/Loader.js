import './Loader.css';
const Loader = ({ cars }) => {
    return ( 
        <div className="loader-list-container list-wrapper">
            <div className="list-header">
                <h3>Available Cars</h3>
                <p className="medium"></p>
            </div>
            {cars.map((car,index)=>(
                <div className="loader-list-item" key={index}>
                    <div className="product-icon"></div>
                    <div className="list-item-content">
                        <p className="medium"></p>
                        <p className="medium"></p>
                        <p className="small"></p>
                    </div>
                </div>
            ))}
            <div className="loader-pagination-container">
                <span className="medium"></span>
            </div>
        </div>
    );
}
 
export default Loader;
