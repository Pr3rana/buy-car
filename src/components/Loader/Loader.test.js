import { render, screen, cleanup, carData } from '../../test-utils.js';
import Loader from './Loader'

afterEach(cleanup)

describe("Loader render",()=>{
    test('Initial render', ()=>{
        render(<Loader />)
        expect(screen.getByTestId('loader-list-container')).toMatchSnapshot(
            `
            <div data-testid="loader-list-container" className="loader-list-container list-wrapper">
                <div className="list-header">
                    <h3>Available Cars</h3>
                    <p className="medium"></p>
                </div>
                <div className="loader-pagination-container">
                    <span className="medium"></span>
                </div>
            </div>
            `
        )
    })
})

describe("Loader props",()=>{
    test("testing loader with props",()=>{
        render(<Loader cars={carData.cars} />)
        expect(screen.getByTestId('loader-list-container')).toMatchSnapshot(
            `<div data-testid="loader-list-container" className="loader-list-container list-wrapper">
                <div className="list-header">
                    <h3>Available Cars</h3>
                    <p className="medium"></p>
                </div>
                <div className="loader-list-item" key={index}>
                    <div className="product-icon"></div>
                    <div className="list-item-content">
                        <p className="medium"></p>
                        <p className="medium"></p>
                        <p className="small"></p>
                    </div>
                </div>
                <div className="loader-pagination-container">
                    <span className="medium"></span>
                </div>
            </div>`
        )
    })
})
