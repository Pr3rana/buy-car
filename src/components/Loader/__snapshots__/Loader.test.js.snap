// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader props testing loader with props: <div data-testid="loader-list-container" className="loader-list-container list-wrapper">
                <div className="list-header">
                    <h3>Available Cars</h3>
                    <p className="medium"></p>
                </div>
                <div className="loader-list-item" key={index}>
                    <div className="product-icon"></div>
                    <div className="list-item-content">
                        <p className="medium"></p>
                        <p className="medium"></p>
                        <p className="small"></p>
                    </div>
                </div>
                <div className="loader-pagination-container">
                    <span className="medium"></span>
                </div>
            </div> 1`] = `
<div
  class="loader-list-container list-wrapper"
  data-testid="loader-list-container"
>
  <div
    class="list-header"
  >
    <h3>
      Available Cars
    </h3>
    <p
      class="medium"
    />
  </div>
  <div
    class="loader-list-item"
  >
    <div
      class="product-icon"
    />
    <div
      class="list-item-content"
    >
      <p
        class="medium"
      />
      <p
        class="medium"
      />
      <p
        class="small"
      />
    </div>
  </div>
  <div
    class="loader-pagination-container"
  >
    <span
      class="medium"
    />
  </div>
</div>
`;

exports[`Loader render Initial render: 
            <div data-testid="loader-list-container" className="loader-list-container list-wrapper">
                <div className="list-header">
                    <h3>Available Cars</h3>
                    <p className="medium"></p>
                </div>
                <div className="loader-pagination-container">
                    <span className="medium"></span>
                </div>
            </div>
             1`] = `
<div
  class="loader-list-container list-wrapper"
  data-testid="loader-list-container"
>
  <div
    class="list-header"
  >
    <h3>
      Available Cars
    </h3>
    <p
      class="medium"
    />
  </div>
  <div
    class="loader-pagination-container"
  >
    <span
      class="medium"
    />
  </div>
</div>
`;
